### PR TITLE
ruby_execute

### PR DESCRIPTION
**Pull request by tylercloke**
### Notes
If these commands are (almost) exclusively for use in delivery, we should consider adding some magic whereby paths to C things get set for bundler to `chef_execute`. I've had to pass things like:

```
"LDFLAGS" => "-Wl,-rpath,#{install_dir}/embedded/lib -L#{install_dir}/embedded/lib",
"CFLAGS" => "-I#{install_dir}/embedded/include"
```

to bundle commands in delivery to get things to work. Not sure on all the details, just something to investigate.

### TODO
- [ ] `libssl-dev` on all unix platforms.
- [ ] `*_install` per comments.
- [ ] `kitchen converge` works windows but `kitchen verify` does not yet. Not sure what is expected.

### CODE REVIEW

File: libraries/ruby_install.rb [hunk 1]
```diff
@@ -96,11 +98,42 @@ def installed?
...
+        install_package('libxml2-dev')
+        install_package('libxslt-dev')
+        install_package('zlib1g-dev')
+        install_package('ncurses-dev')
+        install_package('libssl-dev')
+      elsif freebsd?
+        install_package('textproc/libxml2')
+        install_package('textproc/libxslt')
+        install_package('devel/ncurses')
+        install_package('libssl-dev')
```
Comment (tylercloke) [inline libraries/ruby_install.rb]: TODO make sure `libssl-dev` is correct for non-debian platforms.

File: libraries/ruby_install.rb [hunk 2]
```diff
@@ -96,11 +98,42 @@ def installed?
...
+        install_package('libxml2')
+        install_package('libxslt')
+        install_package('libssl-dev')
+      elsif rhel?
+        install_package('libxml2-devel')
+        install_package('libxslt-devel')
+        install_package('ncurses-devel')
+        install_package('zlib-devel')
+        install_package('libssl-devel')
+      end
```
Comment (schisamo) [inline libraries/ruby_install.rb]: I wonder if all these base packages should be installed by a helper method shared by all `*_install` resources?

File: libraries/ruby_install.rb [hunk 3]
```diff
@@ -111,6 +144,11 @@ def install_dependencies
       ruby_install.run_action(:install)
     end
 
+    def install_package(package_str)
+      pkg = Chef::Resource::Package.new(package_str, run_context)
+      pkg.run_action(:install)
+    end
```
Comment (schisamo) [inline libraries/ruby_install.rb]: This also feels like a helper shared by all `*_install` resources.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

File: libraries/ruby_install.rb (hunk 2 above)
Comment (tylercloke) [inline libraries/ruby_install.rb]: Yeah, that is in my TODO at the start of the block. Probably a good call. I was gonna wait until we had a common set and abstract them for all language install recipes.

**Comment by schisamo** [comment]
We should have a base `LanguageExecute` resource that things like `RubyExecute` and `NodeExecute` extend from. All execute resources are going to share most all attributes: `command`, `version`, `prefix`, `cwd` etc. That way all resources share the same validation logic etc.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by tylercloke** [comment]
:+1:

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by chef-delivery** [comment]
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

**Comment by schisamo** [comment]
@delivery approve

**Comment by chef-delivery** [comment]
Change: ae25470d-91d8-4cc7-96f9-aa1c41bcfdcb approved by: @schisamo

**Comment by chef-delivery** [comment]
This PR has passed 'Acceptance' and is ready to be delivered!
  Use: '@delivery deliver' when validated in acceptance.

**Pull request merged**